src/connection.js

```diff
@@ -392,7 +393,23 @@ export default class Connection extends EventEmitter {
     if (this.config.options.localAddress) {
       connectOpts.localAddress = this.config.options.localAddress;
     }
-    this.socket.connect(connectOpts);
+    
+    if (!multiSubnetFailover) {
+        this.socket.connect(connectOpts);
+    }
+    else {
+      //look up both ipv4 and ipv6 addresses
+      this.resolveIPaddresses(connectOpts.host, (addresses) => {
+        if(typeof addresses !== 'undefined' && addresses.length > 1) {
```

Comment:
I want to keep check of ```addresses.length >1``` here. If there is only one ip address for a host, we should fall back to regular connect logic